spyder/utils/tests/test_sourcecode.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+
+"""Tests for sourcecode.py"""
+
+import os
+import sys
+
+import pytest
+
+from spyder.utils import sourcecode
+
+
+def test_get_primary_at(tmpdir, qtbot):
+    code = 'import functools\nfunctools.partial'
+    assert sourcecode.get_primary_at(code, len(code)) == 'functools.partial'
+
+
+def test_get_identifiers(tmpdir, qtbot):
+    code = 'import functools\nfunctools.partial'
+    assert set(sourcecode.get_identifiers(code)) == set(['import', 'functools',
+                                              'functools.partial'])
+
+
+def test_split_source(tmpdir, qtbot):
+    code = 'import functools\nfunctools.partial'
+    assert sourcecode.split_source(code) == ['import functools', 'functools.partial']
+    code = code.replace('\n', '\r\n')
+    assert sourcecode.split_source(code) == ['import functools', 'functools.partial']
+
+
+def test_path_components(tmpdir, qtbot):
+    if sys.platform.startswith('linux'):
+        path_components0 = ['','','documents','test','test.py']
+        path_components1 = ['','','documents','projects','test','test.py']
```

Comment:
Why do we need to test two `path_components` here? Is not enough with testing only one?